### PR TITLE
Rnn `train` simplification

### PR DESCRIPTION
**Pull request by robertleeplummerjr**
This simplifies the training process for RNN's to be very close to that of what is already supported in brain.js.

### CODE REVIEW

**Comment by robertleeplummerjr** [comment]
Here are two examples:
```js
      const net = new LSTM({
        //json: json
      });
      const trainingData = [{
        input: 'hi',
        output: 'mom'
      }, {
        input: 'howdy',
        output: 'dad'
      }, {
        input: 'hello',
        output: 'sis'
      }, {
        input: 'yo',
        output: 'bro'
      }];
      net.train(trainingData, { iterations: 100, log: true });
      assert.equal(net.run('hi'), 'mom');
      assert.equal(net.run('howdy'), 'dad');
      assert.equal(net.run('hello'), 'sis');
      assert.equal(net.run('yo'), 'bro');
    });
```
and
```js
      const net = new LSTM();
      const transationTypes = {
        credit: 0,
        debit: 1,
        personalCard: 2,
        other: 3
      };
      const trainingData = [{
        input: [transationTypes.credit],
        output: 'credit'
      }, {
        input: [transationTypes.debit],
        output: 'debit'
      }, {
        input: [transationTypes.personalCard],
        output: 'personal card'
      }, {
        input: [transationTypes.other],
        output: 'other'
      }];
      net.train(trainingData, { iterations: 100, log: true });
      assert.equal(net.run([transationTypes.credit]), 'credit');
      assert.equal(net.run([transationTypes.debit]), 'debit');
      assert.equal(net.run([transationTypes.personalCard]), 'personal card');
      assert.equal(net.run([transationTypes.other]), 'other');
```

**Comment by Dok11** [comment]
Why not in a wiki?

File: src/recurrent/rnn.js [hunk 1]
```diff
@@ -38,6 +42,12 @@ export default class RNN {
       outputMatrixIndex: -1
     };
 
+    if (this.vocab !== null) {
+      this.inputSize = this.vocab.characters.length;
+      this.inputRange = this.vocab.characters.length;
+      this.outputSize = this.vocab.characters.length;
```
Comment (IonicaBizau) [inline src/recurrent/rnn.js:48]: Wouldn't it be better to do it in one line?

```js
this.inputSize = this.inputRange = this.outputSize = this.vocab.characters.length;
```

File: src/utilities/vocab.js [hunk 2]
```diff
@@ -6,28 +6,41 @@
  */
 export default class Vocab {
   constructor(values, maxThreshold = 0) {
+    if (typeof values === 'undefined') return;
```
Comment (IonicaBizau) [inline src/utilities/vocab.js]: Maybe `values === undefined` or `values === void 0`?

**Comment by robertleeplummerjr** [comment]
I like the way you think!

**Review by IonicaBizau — COMMENTED** [review]
Didn't find anything wrong, but just left some syntax-related suggestions.

**Comment by robertleeplummerjr** [comment]
ty @IonicaBizau!  Updated.

**Comment by IonicaBizau** [comment]
:+1:

**Pull request merged**